weblate/trans/views/settings.py

```diff
@@ -175,6 +175,17 @@ def dismiss_alert(request: AuthenticatedHttpRequest, path):
     return redirect_param(obj, "#alerts")
 
 
+def flag_categories_to_delete(categories):
+    while len(categories) > 0:
+        category = categories.pop()
+        category.to_delete = True
+        category.save()
+        categories += category.category_set.all()
+        for component in category.component_set.all():
+            component.to_delete = True
+            component.save()
```

Comment:
This can be handled more effectively by using `update()`, see below.

Reply:
Optimized this function using `update()` and removing the loop over components.